cloudinit/net/sysconfig.py

```diff
@@ -466,10 +487,17 @@ def _render_subnet_routes(cls, iface_cfg, route_cfg, subnets):
                     # TODO(harlowja): add validation that no other iface has
                     # also provided the default route?
                     iface_cfg['DEFROUTE'] = True
+                    # TODO(hjensas): Including dhcp6 here is likely incorrect.
+                    # DHCPv6 does not ever provide a default gateway, the
+                    # default gateway come from RA's.
+                    # (https://github.com/openSUSE/wicked/issues/570)
                     if iface_cfg['BOOTPROTO'] in ('dhcp', 'dhcp4', 'dhcp6'):
+                        # NOTE(hjensas): DHCLIENT_SET_DEFAULT_ROUTE is SuSE
+                        # only. RHEL, CentOS, Fedora does not implement this
+                        # option.
                         iface_cfg['DHCLIENT_SET_DEFAULT_ROUTE'] = True
                     if 'gateway' in route:
-                        if is_ipv6 or is_ipv6_addr(route['gateway']):
+                        if is_ipv6:
```

Comment:
Won't we miss in a purely static ipv6 configuration?

Reply:
is_ipv6 is true if route['gateway'] is an IPv6 address, see my comment on L466 above.